File: INTRODUCCION-POO/EJERCICIO-9/Computadora.py
# INTRODUCCIÓN A LA PROGRAMACIÓN ORIENTADA A OBJETOS EN PYTHON


class Computadora:
    
    def __init__(self, marca, cpu, ram, almacenamiento):
        self.marca = marca            
        self.cpu = cpu              
        self.ram = ram                
        self.almacenamiento = almacenamiento  
        self.encendida = False       

    
    def mostrar_componentes(self):
        print(f"Componentes de la computadora {self.marca}:")
        print(f"CPU: {self.cpu}")
        print(f"RAM: {self.ram} GB")
        print(f"Almacenamiento: {self.almacenamiento} GB")

    # Método para encender la computadora
    def encender(self):
        if not self.encendida:
            self.encendida = True
            print(f"La computadora {self.marca} está encendida.")
        else:
            print(f"La computadora {self.marca} ya está encendida.")

    # Método para apagar la computadora
    def apagar(self):
        if self.encendida:
            self.encendida = False
            print(f"La computadora {self.marca} está apagada.")
        else:
            print(f"La computadora {self.marca} ya está apagada.")

    # Método para mostrar el estado de la computadora
    def mostrar_estado(self):
        estado = "encendida" if self.encendida else "apagada"
        print(f"La computadora {self.marca} está {estado}.")


# Crear una instancia de la clase Computadora
mi_computadora = Computadora("Dell", "Intel i7", 16, 512)

mi_computadora.mostrar_componentes()

# Simular encender y apagar la computadora
mi_computadora.encender()
mi_computadora.mostrar_estado()

mi_computadora.apagar()
mi_computadora.mostrar_estado()

File: INTRODUCCION-POO/EJERCICIO-9/Computadora.java
// INTRODUCCIÓN A LA PROGRAMACIÓN ORIENTADA A OBJETOS EN JAVA
//Realiza la abstracción de una Computadora
//a) Muestra los componentes principales de la computadora
//b) Muestra el estado de la computadora (encendido o apagado)
//c) Crea una instancia y simula encender y apagar la computadora.
class Computadora {
    
    private String marca;
    private String cpu;
    private int ram;
    private int almacenamiento;
    private boolean encendida;

    
    public Computadora(String marca, String cpu, int ram, int almacenamiento) {
        this.marca = marca;
        this.cpu = cpu;
        this.ram = ram;
        this.almacenamiento = almacenamiento;
        this.encendida = false;  
    }

    
    public void mostrarComponentes() {
        System.out.println("Componentes de la computadora " + this.marca + ":");
        System.out.println("CPU: " + this.cpu);
        System.out.println("RAM: " + this.ram + " GB");
        System.out.println("Almacenamiento: " + this.almacenamiento + " GB");
    }

    // Método para encender la computadora
    public void encender() {
        if (!this.encendida) {
            this.encendida = true;
            System.out.println("La computadora " + this.marca + " está encendida.");
        } else {
            System.out.println("La computadora " + this.marca + " ya está encendida.");
        }
    }

    // Método para apagar la computadora
    public void apagar() {
        if (this.encendida) {
            this.encendida = false;
            System.out.println("La computadora " + this.marca + " está apagada.");
        } else {
            System.out.println("La computadora " + this.marca + " ya está apagada.");
        }
    }

    // Método para mostrar el estado de la computadora
    public void mostrarEstado() {
        String estado = this.encendida ? "encendida" : "apagada";
        System.out.println("La computadora " + this.marca + " está " + estado + ".");
    }

    
    public static void main(String[] args) {
        
        Computadora miComputadora = new Computadora("Dell", "Intel i7", 16, 512);

        
        miComputadora.mostrarComponentes();

        // Simular encender y apagar la computadora
        miComputadora.encender();
        miComputadora.mostrarEstado();

        miComputadora.apagar();
        miComputadora.mostrarEstado();
    }
}